speechbrain/lobes/models/wav2vec.py

```diff
@@ -0,0 +1,610 @@
+import math
+import logging
+import torch
+import torch.nn.functional as F
+import torch.nn as nn
+import random
+import numpy as np
+from contextlib import contextmanager
+import time
+
+from speechbrain.lobes.models.transformer.Transformer import (
+    TransformerEncoder,
+    PositionalEncoding,
+)
+from speechbrain.utils.data_utils import batch_pad_right
+from torch.utils.data._utils.collate import default_convert
+from speechbrain.utils.data_utils import mod_default_collate
+from speechbrain.utils.data_utils import recursive_to
+from torch.utils.data._utils.pin_memory import (
+    pin_memory as recursive_pin_memory,
+)
+import collections
+from speechbrain.lobes.models.convolution import ConvolutionFrontEnd
+from speechbrain.nnet.CNN import Conv1d
+from speechbrain.nnet.normalization import LayerNorm
+from speechbrain.nnet.quantisers import GumbelVectorQuantizer
+from speechbrain.dataio.batch import PaddedBatch
+
+logger = logging.getLogger()
+
+
+class W2VLatentExtractor(nn.Module):
+    """Convolution based feature extractor from raw audio.
+
+    Channel numbers increasing is based on https://arxiv.org/abs/2109.06870
+    """
+
+    def __init__(
+        self,
+        out_channels=[512, 512, 512, 512, 512, 512, 512],
+        kernel_sizes=[11, 3, 3, 3, 3, 3, 3],
+        strides=[5, 2, 2, 2, 2, 2, 2],
+        dropout=0.0,
+        init='kaiming',
+        extractor_mode="layer_norm",
+    ):
+        super().__init__()
+        if extractor_mode != "layer_norm":
```

Comment:
What is the point of having a parameter if we only have one implementation ?

Reply:
Since there are other wav2vec2.0 variants that use Group Norm, we use that argument as a placeholder. But I agree that maybe we could just erase that for simplicity, until we finally implement Group Norm or other variants.

Reply:
yea this is kind of silly you're right

Reply:
done